test/e2e/shared/common.go

```diff
@@ -199,15 +202,15 @@ func dumpMachine(ctx context.Context, e2eCtx *E2EContext, machine infrav1.OpenSt
 			},
 			{
 				title: "containerd-info",
-				cmd:   "crictl info",
+				cmd:   "crictl --runtime-endpoint unix:///run/containerd/containerd.sock info",
 			},
 			{
 				title: "containerd-containers",
-				cmd:   "crictl ps",
+				cmd:   "crictl --runtime-endpoint unix:///run/containerd/containerd.sock ps",
 			},
 			{
 				title: "containerd-pods",
-				cmd:   "crictl pods",
+				cmd:   "crictl --runtime-endpoint unix:///run/containerd/containerd.sock pods",
```

Comment:
We are using containerd in all our tests. Previously we were relying on `crictl` checking all default locations and finding containerd that way, but if you check the logs from any recent runs you will see that it no longer works. It is now erroring out because we do not specify the endpoint. I'm not sure when exactly this changed.

Reply:
Here is an example where you an see how it currently fails: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1590/pull-cluster-api-provider-openstack-e2e-test/1668820465856024576#1:build-log.txt%3A2358
and as a result we do not get the corresponding logs in the artifacts: https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1590/pull-cluster-api-provider-openstack-e2e-test/1668820465856024576/artifacts/clusters/bootstrap/instances/e2e-1ydzzp/cluster-e2e-1ydzzp-control-plane-4bx9j/

Reply:
Crictl has this note about the endpoint:

> Note: The default endpoints are now deprecated and the runtime endpoint should always be set instead. The performance maybe affected as each default connection attempt takes n-seconds to complete before timing out and going to the next in sequence.

From https://github.com/kubernetes-sigs/cri-tools/blob/master/docs/crictl.md#usage